packages/console/src/components/Select/index.tsx

```diff
@@ -0,0 +1,83 @@
+import classNames from 'classnames';
+import React, { ReactNode, useLayoutEffect, useRef, useState } from 'react';
+
+import { ArrowDown, ArrowUp } from '@/icons/Arrow';
+
+import Dropdown, { DropdownItem } from '../Dropdown';
+import * as styles from './index.module.scss';
+
+type Option = {
+  value: string;
+  title: ReactNode;
+};
+
+type Props = {
+  value?: string;
+  options: Option[];
+  onChange?: (value: string) => void;
+  isReadOnly?: boolean;
+  hasError?: boolean;
+};
+
+const Select = ({ value, options, onChange, isReadOnly, hasError }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const anchorRef = useRef<HTMLInputElement>(null);
+  const current = options.find((option) => value && option.value === value);
+  const [width, setWidth] = useState<number>();
+
+  const handleSelect = (value: string) => {
+    onChange?.(value);
+    setIsOpen(false);
+  };
+
+  useLayoutEffect(() => {
+    if (anchorRef.current) {
+      const { width } = anchorRef.current.getBoundingClientRect();
+      setWidth(width);
```

Comment:
Do we need this useLayoutEffect and setState? Can we directly use the `ref.current` as the width?

Reply:
Interesting, but then it may be executed before render (dom ready)?

Reply:
Let me try it

Reply:
Per offline discussion, changed to the initial implementation.